xbmc/network/upnp/UPnPInternal.cpp

```diff
@@ -141,12 +147,28 @@ GetMimeType(const CFileItem& item,
 
     NPT_String mime;
 
-    /* We always use Platinum mime type first
-       as it is defined to map extension to DLNA compliant mime type
-       or custom according to context (who asked for it) */
-    if (!ext.IsEmpty()) {
-        mime = PLT_MimeType::GetMimeTypeFromExtension(ext, context);
-        if (mime == "application/octet-stream") mime = "";
+    /* We look first to our extensions/overrides of libplatinum mimetypes. If not found, fallback to
+     Platinum definitions.  */
+    if (!ext.IsEmpty())
+    {
+      for (const NPT_HttpFileRequestHandler_DefaultFileTypeMapEntry& mimeTypeEntry :
+           KodiPlatinumMimeTypeExtensions)
+      {
+        if (mimeTypeEntry.extension == ext)
+        {
+          mime = mimeTypeEntry.mime_type;
+          break;
+        }
+      }
+    }
+
+    /* Give priority to Platinum mime types as they are defined to map extension to DLNA compliant mime types
+       or custom types according to context (who asked for it) */
+    if (!ext.IsEmpty() && mime.IsEmpty())
+    {
+      mime = PLT_MimeType::GetMimeTypeFromExtension(ext, context);
+      if (mime == "application/octet-stream")
+        mime = "";
```

Comment:
Worthwhile just dropping the isempty check and putting this block inside the previous isempty check after the for loop?